### PR TITLE
User story 38 - Merchant admin can add new items

### PR DESCRIPTION
**Pull request by nancylee713**
- Only merchant admin can add new items 
  - `Add New Item` is displayed only on merchant/items page
  - regular user cannot randomly add items
  - remove a spec test file (`items/new_spec.rb`), but we still need to keep #new and #create in items controller for merchant admin to add new items

- Modify validations for item attributes to match this user story's requirements
- RSpec OK

- Note: I may need to come back to `add_item_spec.rb` test to verify contents for `enabled?` and 'active?'

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -5,13 +5,11 @@ class Item <ApplicationRecord
...
-  validates_presence_of :name,
-                        :description,
-                        :price,
-                        :image,
-                        :inventory
+  validates_presence_of :name, :description, :price, :inventory
+  validates :image, presence: true, allow_blank: true
   validates_inclusion_of :active?, :in => [true, false]
   validates_numericality_of :price, greater_than: 0
+  validates_numericality_of :inventory, greater_than_or_equal_to: 0
```
Comment (tschaffer1618) [inline app/models/item.rb]: Should we add the only_integer validation for inventory as well?

Comment (leiyakenney) [reply app/models/item.rb]: Ooh I like that idea! That was one of the things I had written down as feedback from Little Shop

File: app/controllers/items_controller.rb [hunk 2]
```diff
@@ -18,10 +19,16 @@ def new
   end
 
   def create
-    @merchant = Merchant.find(params[:merchant_id])
+    @merchant = current_user.merchant
```
Comment (leiyakenney) [inline app/controllers/items_controller.rb]: Nice!

File: app/controllers/merchant/items_controller.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 class Merchant::ItemsController < Merchant::BaseController
   def index
-    @items = Item.where(merchant_id: current_user.merchant_id)
+    @merchant = current_user.merchant
+    @items = current_user.merchant.items    
   end
```
Comment (leiyakenney) [inline app/controllers/merchant/items_controller.rb]: Thanks for updating these to be more what Brian is looking for

File: spec/models/item_spec.rb [hunk 4]
```diff
@@ -4,9 +4,9 @@
   describe "validations" do
     it { should validate_presence_of :name }
     it { should validate_presence_of :description }
-    it { should validate_presence_of :price }
-    it { should validate_presence_of :image }
-    it { should validate_presence_of :inventory }
+    it { should allow_value(nil).for(:image) }
+    it { should validate_numericality_of :price }
+    it { should validate_numericality_of :inventory }
     it { should validate_inclusion_of(:active?).in_array([true,false]) }
```
Comment (leiyakenney) [inline spec/models/item_spec.rb]: Love the numericality validation here

**Pull request merged**